internal/wallet/utxo.go

```diff
@@ -0,0 +1,191 @@
+package wallet
+
+import (
+	"encoding/hex"
+
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/btcsuite/btcwallet/wtxmgr"
+)
+
+// Utxo is a unspend transaction output
+type Utxo btcjson.ListUnspentResult
+
+// ListUnspent returns unspent transactions.
+// TODO: add filter
+//   Only utxos with address contained the param addresses will be considered.
+//   If param addresses is empty, all addresses are considered and there is no
+//   filter
+func (w *Wallet) ListUnspent() ([]*btcjson.ListUnspentResult, error) {
+	var results []*btcjson.ListUnspentResult
+	err := walletdb.View(w.db, func(tx walletdb.ReadTx) error {
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+		txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
+
+		syncBlock := w.Manager.SyncedTo()
+		// filter := len(addresses) != 0
+
+		unspent, e := w.TxStore.UnspentOutputs(txmgrNs)
+		if e != nil {
+			return e
+		}
+
+		// make btcjson.ListUnspentResult from Credit types
+		results = make([]*btcjson.ListUnspentResult, 0, len(unspent))
+		for i := range unspent {
+			output := unspent[i]
+			result := w.credit2ListUnspentResult(output, syncBlock, addrmgrNs)
+			// TODO: result might return nil... catch that nil?
+			results = append(results, result)
+		}
+		return nil
+	})
+	return results, err
+}
+
+func (w *Wallet) credit2ListUnspentResult(c wtxmgr.Credit, syncBlock waddrmgr.BlockStamp,
```

Comment:
Getting a lint error here... line 49 is 89 characters long. I'm not sure what's the best way to structure func heading